packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -100,7 +142,14 @@ export const NotificationsContextProvider = ({
         return !!globalThis.window?.Notification;
       },
     }),
-    [hasPermission, currentUnreadCount, isInitialized, user],
+    [
+      hasPermission,
+      currentUnreadCount,
+      isInitialized,
+      user,
+      hasEnabledPermission,
+      hasPermissionCache,
```

Comment:
I find the naming here somewhat confusing. Not sure what's the difference between `hasPermission`, `hasEnabledPermission`, `hasPermissionCache`